tests/app/views/test_frameworks.py

```diff
@@ -202,62 +200,169 @@ class TestUpdateFramework(BaseApplicationTest, JSONUpdateTestMixin):
+            'lots': ['saas', 'paas', 'iaas', 'scs']
+        }
 
-    def teardown(self):
-        with self.app.app_context():
-            Framework.query.filter(Framework.slug == 'example').delete()
-            db.session.commit()
+        self.attribute_whitelist = [
+            'frameworkAgreementDetails',
+            'status',
+            'clarificationQuestionsOpen',
+        ]
 
-    def test_framework_updated(self):
+    def post_framework_update(self, update):
+        return self.client.post(
+            '/frameworks/example-framework',
+            data=json.dumps({
+                'frameworks': update,
+                'updated_by': 'example user'
+            }),
+            content_type="application/json"
+        )
+
+    def test_returns_404_on_non_existent_framework(self, open_example_framework):
         with self.app.app_context():
-            response = self.client.post('/frameworks/example',
-                                        data=json.dumps({'frameworks': {
-                                            'status': 'expired',
-                                            'clarificationQuestionsOpen': False,
-                                        }, 'updated_by': 'example user'}),
-                                        content_type="application/json")
+            response = self.client.post(
+                '/frameworks/example-framework-2',
+                data=json.dumps({'frameworks': {
+                    'status': 'expired'
+                }, 'updated_by': 'example user'}),
+                content_type="application/json"
+            )
 
-            assert response.status_code == 200
+            assert response.status_code == 404
 
-            framework = Framework.query.filter(Framework.slug == 'example').first()
-            assert framework.status == "expired"
-            assert not framework.clarification_questions_open
+    def test_can_update_whitelisted_fields(self, open_example_framework):
+        valid_attributes_and_values = {
+            key: value for key, value in self.framework_attributes_and_values_for_update.items()
+            if key in self.attribute_whitelist
+        }
 
-    def test_returns_404_on_non_existent_framework(self):
         with self.app.app_context():
-            response = self.client.post('/frameworks/example2',
-                                        data=json.dumps({'frameworks': {'status': 'expired'},
-                                                         'updated_by': 'example user'}),
-                                        content_type="application/json")
+            for key, value in valid_attributes_and_values.items():
+                response = self.post_framework_update({
+                    key: value
+                })
+
+                assert response.status_code == 200
+                post_data = json.loads(response.get_data())['frameworks']
+
+                # frameworkAgreementDetails doesn't show up in the serialized framework
```

Comment:
Ah I see what you've done here. The comment could be a little clearer though, to show that the contents of frameworkAgreementDetails are being transformed into a different format in the serialize method rather than they are not included.

Feel free to improve but one suggestion could be:
frameworkAgreementDetails is not included in Framework.serialize itself but instead each key, value in  frameworkAgreementDetails are included as separate keys in the return from Framework.serialize